compiler-plugin/src/main/kotlin/com/google/devtools/ksp/KotlinSymbolProcessingExtension.kt

```diff
@@ -107,7 +107,11 @@ abstract class AbstractKotlinSymbolProcessingExtension(val options: KspOptions,
         bindingTrace: BindingTrace,
         files: Collection<KtFile>
     ): AnalysisResult? {
-        return AnalysisResult.success(BindingContext.EMPTY, module, shouldGenerateCode = false)
+        return if (logger.errorCount > 0 || logger.exceptionCount > 0) {
+            AnalysisResult.success(bindingTrace.bindingContext, module, shouldGenerateCode = false)
+        } else {
+            AnalysisResult.success(BindingContext.EMPTY, module, shouldGenerateCode = false)
+        }
```

Comment:
Just curious, why not simply changing `BindingContext.EMPTY` to `bindingTrace.bindingContext`?

Reply:
Because we don't want to fail processing in this branch.

Reply:
This kind of explanation isn't really helpful. Please provide more information next time. For example, there are other errors from compiler such as symbol not found that are not fatal to ksp.